src/libs/markAllReportsAsRead.ts

```diff
@@ -0,0 +1,34 @@
+import type {OnyxCollection} from 'react-native-onyx';
+import Onyx from 'react-native-onyx';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {Report, ReportActions} from '@src/types/onyx';
+import {readNewestAction} from './actions/Report';
+import {getOneTransactionThreadReportID} from './ReportActionsUtils';
+import {isUnread} from './ReportUtils';
+
+let allReports: OnyxCollection<Report> = {};
+Onyx.connect({
+    key: ONYXKEYS.COLLECTION.REPORT,
+    waitForCollectionCallback: true,
+    callback: (value) => (allReports = value),
+});
+
+let allReportActions: OnyxCollection<ReportActions> = {};
+Onyx.connect({
+    key: ONYXKEYS.COLLECTION.REPORT_ACTIONS,
+    waitForCollectionCallback: true,
+    callback: (value) => (allReportActions = value),
+});
+
+export default function markAllReportsAsRead() {
+    Object.keys(allReports ?? {}).forEach((key: string) => {
+        const report = allReports?.[key];
+        const oneTransactionThreadReportID = getOneTransactionThreadReportID(report?.reportID, allReportActions?.[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${report?.reportID}`]);
+        const oneTransactionThreadReport = allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${oneTransactionThreadReportID}`];
+        if (!isUnread(report, oneTransactionThreadReport)) {
+            return;
+        }
+
+        readNewestAction(report?.reportID);
```

Comment:
In https://github.com/Expensify/App/blob/04e58bdfb72bffebb7511208f442881830207f75/src/libs/markAllPolicyReportsAsRead.ts I noticed that we apply a 1sec delay between each action. Is this done to prevent flooding the server? Shouldn't we do the same here?

Reply:
I don't find any reason why we add a timeout on the [PR](https://github.com/Expensify/App/pull/38194) that adds this. I also thought that it could be to prevent flooding the server, but the timeout keeps increasing. 0s, 1s, 2s, 3s, ... If there is 100 unread report, then it will be delayed by 100s (including the optimistic action). Also, ReadNewestAction is a WRITE, so it gonna wait 1 by 1.

For the comment above, the reason I create a separate file is that I don't want to use the timeout, so it gonna be cleaner, no timeout, no filter.

Reply:
I have asked in the [PR](https://github.com/Expensify/App/pull/38194#discussion_r2082993036). Moving this to the BE makess sense to me. I have also raised this in [slack](https://expensify.slack.com/archives/C01GTK53T8Q/p1746861626079909).